index.html

```diff
@@ -0,0 +1,451 @@
+  query: Query
+}
+
+enum csdl_Graph {
+  ROCKETS    @csdl_endpoint(url: "https://rockets.example.com"),
+  ASTRONAUTS @csdl_endpoint(url: "https://astronauts.example.com"),
+}
+
+type Query {
+  rockets: [Rocket]!       @csdl_resolve(graph: ROCKETS)
+  astronauts: [Astronaut]! @csdl_resolve(graph: ASTRONAUTS)
+}
+
+type Astronaut
+  @csdl_key(graph: ASTRONAUTS, repr: "{ id }")
+  @csdl_key(graph: ROCKETS, repr: "{ id }")
+{
+  id: String!     @csdl_resolve(graph: ASTRONAUTS)
+  name: String!   @csdl_resolve(graph: ASTRONAUTS)
+  tripId: String! @csdl_resolve(graph: ASTRONAUTS)
+  rocket: Rocket! @csdl_resolve(graph: ASTRONAUTS)
+}
+
+type Rocket 
+  @csdl_key(graph: ASTRONAUTS, repr: "{ id }")
+  @csdl_key(graph: ROCKETS, repr: "{ id }")
+{
+  id: String!           @csdl_resolve(graph: ROCKETS)
+  name: String!         @csdl_resolve(graph: ROCKETS)
+  astronaut: Astronaut! @csdl_resolve(graph: ASTRONAUTS)
+  captain: Astronaut    @csdl_resolve(graph: ASTRONAUTS,
+                                      provides: "{ tripId }")
+}
+```
+
+The meaning of these directives is explored the [Directives](#directives) section.
+
+## Actors
+
+```mermaid
+#::[Actors and roles within an example composition pipeline]
+flowchart TB
+    subgraph A [subgraph A]
+      schemaA([schema A])
+      endpointA([endpoint A])
+    end    
+    subgraph B [subgraph B]
+      schemaB([schema B])
+      endpointB([endpoint B])
+    end
+    subgraph C [subgraph C]
+      schemaC([schema C])
+      endpointC([endpoint C])
+    end
+    subgraph Producer
+      Composer
+    end
+    CSDL([CSDL])
+    subgraph Consumer
+      Router
+    end    
+    A-->Composer
+    B-->Composer
+    C-->Composer
+    Composer-->CSDL
+    CSDL-->Router
+    Router-->published([Published Schema])
+    published-->Clients
+    Clients-->Router
+```
+
+- **Producers** generate CSDL. This spec places requirements on csdl producers.
+- **Consumers** consume CSDL. This spec places requirements on csdl consumers.
+- **Composers** or **compilers** are producers which compose subgraph schemas into CSDL. This document places no particular requirements on the composition algorithm, except that it must produce valid csdl.
+- **Routers** are consumers serve CSDL as a GraphQL endpoint. Graph routers differ from standard GraphQL endpoints in that they are not expected to process data or communicate with (non-GraphQL) backend services on their own. Instead, graph routers receive GraphQL requests and service them by performing additional GraphQL requests. This spec provides guidance for implementing routers, but does not require particular implementations of query separation or dispatch.
+
+Other components of the system include:
+- **Endpoints** are running servers which can resolve GraphQL queries against a schema. In this version of the spec, endpoints must be URLs, typically http/https URLs.
+- **Subgraphs** are the GraphQL schemas which were composed to form the CSDL. A subgraph has:
```

Comment:
I really like the term subgraph.

That said, my impression is that the Rover team had a recent discussion around naming for commands that deal with subgraphs and they landed on `partial`. (I don't know the details of why or where one could read this.) We should try to (eventually) align our terminology here, so if we want to call these things subgraphs (which I personally prefer over partials/partial schemas) that probably means we should change Rover's commands? cc @ashleygwilliams @EverlastingBugstopper

Reply:
Ah, @trevor-scheer came up with subgraph a while ago! I prefer it to partial too, because subgraphs aren't "partial" graphs except in the sense that every graph could be a part of some other graph.

Reply:
Right, my point is that while I agree with you that `subgraph` is a better name, we should probably try to unify our naming across projects, so it's worth seeing if Rover is interested in changing their terminology to `subgraph` too.

Reply:
It's more likely that Rover will adopt what we land on here than the other way around. The `partial` token was a working name, but we never loved it. That said, a big part of that approach was to segregate the stuff around the implementing service model and focus later on pushing a subgraph schema a "graph" in the registry.

Reply:
we've recently renamed our subcommands to `graph` and `subgraph`.